apps/language_server/test/source_file_test.exs

```diff
@@ -35,4 +35,548 @@ defmodule ElixirLS.LanguageServer.SourceFileTest do
            ```
            """
   end
+
+  def new(text) do
+    %SourceFile{text: text, version: 0}
+  end
+
+  describe "apply_content_changes" do
+    defp index_of(string, substring) do
```

Comment:
tests and helper functions ported from https://github.com/microsoft/vscode-languageserver-node